test/util/thread.test.cpp

```diff
@@ -129,6 +129,34 @@ TEST(Thread, context) {
     loop.run();
 }
 
+TEST(Thread, pause) {
+    const std::thread::id tid = std::this_thread::get_id();
+
+    RunLoop loop;
+    std::vector<std::unique_ptr<mbgl::AsyncRequest>> requests;
+
+    Thread<TestObject> thread({"Test"}, tid);
+    Thread<TestObject> unpausedThread({"UnpausedTest"}, tid);
+
+    loop.invoke([&] {
+        thread.pause();
+
+        requests.push_back(thread.invokeWithCallback(&TestObject::fn2, [&] (int result) {
+            EXPECT_EQ(tid, std::this_thread::get_id());
+            EXPECT_EQ(result, 1);
+            loop.stop();
+        }));
+
+        requests.push_back(unpausedThread.invokeWithCallback(&TestObject::transferOut, [&] (std::unique_ptr<int> result) {
+            EXPECT_EQ(tid, std::this_thread::get_id());
+            EXPECT_EQ(*result, 1);
+            thread.resume();
```

Comment:
@tmpsantos I just pushed a wip commit here that duplicates some of what you did in https://github.com/mapbox/mapbox-gl-native/pull/8194 so that I could make sure I understood what you meant about calling resume on a different thread than the one that owns the thread object. I see now that my test still passes because the work task impl `packageArgumentsAndCallback` calls the lambdas such that, in my test, the callback is called on the calling (main in this case) thread so the `MBGL_VERIFY_THREAD` check passes.

Anyway, https://github.com/mapbox/mapbox-gl-native/pull/8194 looks good. I will rebase this on top of that once it is merged. Thanks again.